xbmc/platform/android/activity/JNIXBMCSpeechRecognitionListener.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "JNIXBMCSpeechRecognitionListener.h"
+
+#include "CompileInfo.h"
+#include "ServiceBroker.h"
+#include "messaging/ApplicationMessenger.h"
+#include "utils/log.h"
+
+#include <androidjni/Context.h>
+#include <androidjni/jutils-details.hpp>
+
+using namespace jni;
+
+static std::string s_className =
+    std::string(CCompileInfo::GetClass()) + "/interfaces/XBMCSpeechRecognitionListener";
+
+CJNIXBMCSpeechRecognitionListener::CJNIXBMCSpeechRecognitionListener() : CJNIBase(s_className)
+{
+  m_object = new_object(CJNIContext::getClassLoader().loadClass(GetDotClassName(s_className)));
+  m_object.setGlobal();
+
+  add_instance(m_object, this);
+}
+
+void CJNIXBMCSpeechRecognitionListener::RegisterNatives(JNIEnv* env)
+{
+  jclass cClass = env->FindClass(s_className.c_str());
+  if (cClass)
+  {
+    JNINativeMethod methods[] = {
+        {"_onReadyForSpeech", "(Landroid/os/Bundle;)V",
+         reinterpret_cast<void*>(&CJNIXBMCSpeechRecognitionListener::_onReadyForSpeech)},
+        {"_onError", "(I)V", reinterpret_cast<void*>(&CJNIXBMCSpeechRecognitionListener::_onError)},
+        {"_onResults", "(Landroid/os/Bundle;)V",
+         reinterpret_cast<void*>(&CJNIXBMCSpeechRecognitionListener::_onResults)},
+    };
+
+    env->RegisterNatives(cClass, methods, sizeof(methods) / sizeof(methods[0]));
+  }
+}
+
+CJNIXBMCSpeechRecognitionListener::~CJNIXBMCSpeechRecognitionListener()
+{
+  remove_instance(this);
+}
+
+void CJNIXBMCSpeechRecognitionListener::_onReadyForSpeech(JNIEnv* env, jobject thiz, jobject bundle)
+{
+  CLog::Log(LOGDEBUG, "CJNIXBMCSpeechRecognitionListener onReadyForSpeech()");
+
+  CServiceBroker::GetAppMessenger()->PostMsg(TMSG_SPECH_RECOGNITION_READY);
+}
+
+void CJNIXBMCSpeechRecognitionListener::_onError(JNIEnv* env, jobject thiz, jint i)
+{
+  CLog::Log(LOGDEBUG, "CJNIXBMCSpeechRecognitionListener onError(): [{}]", i);
+
+  CServiceBroker::GetAppMessenger()->PostMsg(TMSG_SPECH_RECOGNITION_ERROR, i);
+}
```

Comment:
Looks like the PostMsg you want is (6 arguments)
```
void PostMsg(uint32_t messageId, int param1, int param2, void* payload, std::string strParam, std::vector<std::string> params);
```
theres not a 5 argument function call with the vector<std::string>